flask/app/tasks.py

```diff
@@ -1,50 +1,34 @@
 import os
 from datetime import datetime, timedelta
+from itertools import chain
+
 from . import models
 from .email import send_email
 from .extensions import db
-from sqlalchemy import text
 
 
 def send_email_notification(app):
-    app.logger.info("hello")
     with app.app_context():
+        yesterday = (datetime.now() - timedelta(1)).strftime("%Y-%m-%d")
 
-        app.logger.info("Sending email notification...")
-
-        yesterday = (datetime.now() - timedelta(5)).strftime("%Y-%m-%d")
-
-        analyses = db.session.query(models.Analysis).filter(
-            models.Analysis.requested >= yesterday,
+        datasets = (
+            db.session.query(models.Dataset)
+            .join(models.Dataset.analyses)
+            .filter(models.Analysis.requested >= yesterday)
+            .all()
```

Comment:
You should be able to do the query from the Analysis side like `models.Analysis.query.options(joinedload(models.Analysis.datasets)).filter(...)`

This query does not do what you want as it will filter based on the analysis requested date, but it won't route those analyses to the relationship, so the next line will requery for those analyses for each dataset individually, without regard for the requested date condition above, so you could end up emailing out old analyses.